spec/requests/conversion_hosts_spec.rb

```diff
@@ -45,4 +45,198 @@
+
+    it "supports single conversion host creation" do
+      api_basic_authorize(collection_action_identifier(:conversion_hosts, :create))
+
+      post(api_conversion_hosts_url, :params => sample_conversion_host_from_vm)
+
+      expect(response).to have_http_status(:ok)
+      expect_result_resources_to_include_keys("results", expected_attributes)
+
+      conversion_host_id = response.parsed_body["results"].first["id"]
+      expect(ConversionHost.exists?(conversion_host_id)).to be_truthy
+    end
+
+    it "supports single conversion host creation via action" do
+      api_basic_authorize(collection_action_identifier(:conversion_hosts, :create))
+
+      post(api_conversion_hosts_url, :params => gen_request(:create, sample_conversion_host_from_vm))
+
+      expect(response).to have_http_status(:ok)
+      expect_result_resources_to_include_keys("results", expected_attributes)
+
+      conversion_host_id = response.parsed_body["results"].first["id"]
+      expect(ConversionHost.exists?(conversion_host_id)).to be_truthy
+    end
+
+    it "supports multiple conversion host creation" do
+      api_basic_authorize(collection_action_identifier(:conversion_hosts, :create))
+
+      conversion_hosts = [sample_conversion_host_from_vm, sample_conversion_host_from_host]
+      post(api_conversion_hosts_url, :params => gen_request(:create, conversion_hosts))
+
+      expect(response).to have_http_status(:ok)
+      expect_result_resources_to_include_keys("results", expected_attributes)
+
+      results = response.parsed_body["results"]
+
+      expect(ConversionHost.exists?(results.first["id"])).to be_truthy
+      expect(ConversionHost.exists?(results.last["id"])).to be_truthy
+      expect(results.first["resource_id"]).to eq(vm.id.to_s)
+      expect(results.last["resource_id"]).to eq(host.id.to_s)
+    end
+  end
+
+  context "delete" do
+    let(:conversion_host)             { FactoryGirl.create(:conversion_host) }
+    let(:conversion_host_url)         { api_conversion_host_url(nil, conversion_host) }
+    let(:invalid_conversion_host_url) { api_conversion_host_url(nil, 999_999) }
+
+    it "can delete a conversion host via DELETE" do
+      api_basic_authorize(action_identifier(:conversion_hosts, :delete))
+      delete(conversion_host_url)
+
+      expect(response).to have_http_status(:no_content)
+    end
+
+    it "can delete a conversion host via POST" do
+      api_basic_authorize(action_identifier(:conversion_hosts, :delete, :resource_actions))
+      post(conversion_host_url, :params => gen_request(:delete))
+
+      expect_single_action_result(:success => true, :message => "deleting", :href => conversion_host_url)
+      expect(ConversionHost.exists?(conversion_host.id)).to be_falsey
+    end
+
+    it "will not delete a conversion host unless authorized" do
+      api_basic_authorize
+      delete(conversion_host_url)
+
+      expect(response).to have_http_status(:forbidden)
```

Comment:
Not sure about this myself: shall we verify that the `conversion_host` is intact?

Reply:
Done.